### ISSUE TITLE
scroll with mouse wheel crashes the app under linux

### ISSUE THREAD

**Issue by Palatis** [telegramdesktop/tdesktop#26205]
### Steps to reproduce

1. launch telegram-desktop
2. scroll with mouse wheel anywhere

### Expected behaviour

scrolls fine

### Actual behaviour

program crash

### Operating system

Ubuntu 23.04 with Gnome 44.0

### Version of Telegram Desktop

4.8

### Installation source

Snap

### Logs

```shell
https://gist.github.com/Palatis/b19f729e4d71fe114dc4949cc16385f3
```

**Comment by Palatis** [comment]
works fine with the version downloaded from [Telegram Desktop](https://updates.tdesktop.com/tlinux/tsetup.4.8.0.tar.xz)

**Comment by Sourgin** [comment]
Same here, just updated and crash every time I scroll.

Downloading the version Palatis mentioned works like a charm now.

**Comment by blain1972** [comment]
I am also affected by this issue Ubuntu 23.04 Telegram 4.8 (snap). However I have just logged into Xorg session and scrolling works fine there.

**Comment by UnityAddict** [comment]
Same issue here with Ubuntu 23.04 Telegram 4.8 snap.

**Comment by artem** [comment]
Same issue with telegram-desktop 4.8.0. Reverting to 4.6.5 helped:

> $ snap revert telegram-desktop
> telegram-desktop reverted to 4.6.5

**Comment by jeffgoh** [comment]
4.8.0 crashes for me - trying 4.8.1
```
snap-id:      jpdoHqMd90M39DdH2JSZbeQxipqHtwLo
tracking:     latest/stable
refresh-date: 2 days ago, at 03:55 +08
channels:
  latest/stable:    4.8.1               2023-04-24 (4759) 408MB -
  latest/candidate: ↑                                           
  latest/beta:      ↑                                           
  latest/edge:      4.8.1-32-g3da787791 2023-04-24 (4756) 408MB -
installed:          4.8.0                          (4743) 408MB -
```

**Comment by ilya-fedin** [comment]
There's no fix in 4.8.1 and talking honestly there's no idea how to fix this problem. There was an idea to rollback the snap to 4.6.5 in the store but we need to know its revision number and sadly no one has helped with that. Now that 4.8.1 is released, 4.6.5 will apparently be removed from all the users' computers by snapd and rollback this will be no longer possible. I honestly have no idea what to do next, probably asking users to switch out of snap package.

**Comment by jeffgoh** [comment]
Yes, @ilya-fedin - upgrading to 4.8.1 was a bad idea - crashes as well and not sure when, but snap wiped out 4.6.5 both on my computer and in the repository, so I can't even revert :(

Based on what @blain1972 says, I could switch from Wayland back to Xorg - but that's not a long term solution at all. Gonna try other packaging methods (rpmfusion, flathub - now that Fedora has unblocked all of flathub, maybe that's the way to go)

**Comment by ilya-fedin** [comment]
Does setting `QT_WAYLAND_RECONNECT=1` help?

**Comment by jeffgoh** [comment]
I didn't try. Switched to flatpak, since I'm on Fedora which opened up flathub for F38. Also snap is becoming a legacy format - being dropped not only by Fedora but also by Canonical (Ubuntu) - https://news.itsfoss.com/ubuntu-ditch-snap/
Will replace my other snap apps and standardize on flatpak if I can (one less packaging subsystem to deal with).

**Comment by ilya-fedin** [comment]
> https://news.itsfoss.com/ubuntu-ditch-snap/

![image](https://user-images.githubusercontent.com/17829319/234143462-5938fd92-bfe8-44d4-a643-92d3e48f7db6.png)

> PS @ilya-fedin scrolling not a problem with Flatpak version. Not sure if your distro supports it.

Not quite sure what you mean by this, I'm not a user asking for help, I'm a contributor.

**Comment by landswim** [comment]
> Does setting `QT_WAYLAND_RECONNECT=1` help?

not work. 
errors:
❯ QT_WAYLAND_RECONNECT=1 telegram-desktop
[ALSOFT] (EE) Failed to connect PipeWire event context (errno: 112)
/usr/share/libdrm/amdgpu.ids: Permission denied
QPainter::begin: Paint device returned engine == 0, type: 2
QWidget::render: Cannot render with an inactive painter
qt.svg: Error while inflating gzip file: SVG format check failed
qt.svg: Error while inflating gzip file: SVG format check failed
qt.svg: Error while inflating gzip file: SVG format check failed
qt.svg: Error while inflating gzip file: SVG format check failed
qt.svg: Error while inflating gzip file: SVG format check failed
The Wayland connection experienced a fatal error: Resource temporarily unavailable
The Wayland connection experienced a fatal error: Resource temporarily unavailable

**Comment by landswim** [comment]
> Folks, could you please provide the revision number of 4.6.5 version?

@ilya-fedin revison is 4654

**Comment by ilya-fedin** [comment]
@landswim thanks, but I guess it's too late at that point. The user data of 4.6.5 was deleted, so pushing this old revision to stable channel will lead to user data being resetted.

**Comment by ilya-fedin** [comment]
It looks like we're in a dead end.

**Comment by rezalabbafi** [comment]
the same Issue telegram Desktop version 4.8 
OS Ubuntu 22.10

**Comment by ansdor** [comment]
happens to me on openSUSE tumbleweed, running KDE/wayland

**Comment by ilya-fedin** [comment]
Does it happen on edge channel?

**Comment by blain1972** [comment]
> Does it happen on edge channel?

it does not - seems to work fine :smile:

**Comment by ilya-fedin** [comment]
Propagated to stable, does stable work now?

**Comment by blain1972** [comment]
> Propagated to stable, does stable work now?

yes :smile:
